src/main/java/org/dita/dost/module/reader/AbstractReaderModule.java

```diff
@@ -51,7 +52,7 @@
 
     Predicate<String> formatFilter;
     /** FileInfos keyed by src. */
-    private final Map<URI, FileInfo> fileinfos = new HashMap<>();
+    private final Map<URI, Collection<FileInfo>> fileinfos = new HashMap<>();
```

Comment:
Instead of mapping _some_ `FileInfo` here by source URI, the value is a list of all `FileInfo` objects that has this source URI.